src/c++/perf_analyzer/client_backend/mock_client_backend.h

```diff
@@ -192,22 +205,43 @@ class MockClientBackend : public ClientBackend {
       const std::vector<InferInput*>& inputs,
       const std::vector<const InferRequestedOutput*>& outputs) override
   {
+    stats_->num_active_infer_calls++;
+
     stats_->CaptureRequest(
         MockClientStats::ReqType::ASYNC, options, inputs, outputs);
-    InferResult* result = new MockInferResult(options);
 
-    callback(result);
+    std::thread([this, &options, callback]() {
+      std::this_thread::sleep_for(stats_->request_delay);
+
+      InferResult* result = new MockInferResult(options);
+      callback(result);
+
+      stats_->num_active_infer_calls--;
+    })
+        .detach();
+
     return Error::Success;
   }
 
   Error AsyncStreamInfer(
       const InferOptions& options, const std::vector<InferInput*>& inputs,
       const std::vector<const InferRequestedOutput*>& outputs)
   {
+    stats_->num_active_infer_calls++;
+
     stats_->CaptureRequest(
         MockClientStats::ReqType::ASYNC_STREAM, options, inputs, outputs);
-    InferResult* result = new MockInferResult(options);
-    stream_callback_(result);
+
+    std::thread([this, &options]() {
+      std::this_thread::sleep_for(stats_->request_delay);
+
+      InferResult* result = new MockInferResult(options);
+      stream_callback_(result);
+
+      stats_->num_active_infer_calls--;
+    })
```

Comment:
These lines look very similar to L213-L222 can they be refactored into a common function?

Reply:
Refactored 👍